plugins/webp-uploads/helper.php

```diff
@@ -406,3 +406,61 @@ function webp_uploads_is_picture_element_enabled(): bool {
 function webp_uploads_is_jpeg_fallback_enabled(): bool {
 	return (bool) get_option( 'perflab_generate_webp_and_jpeg' );
 }
+
+/**
+ * Retrieves the image URL for a specified MIME type from the attachment metadata.
+ *
+ * This function attempts to locate an alternate image source URL in the
+ * attachment's metadata that matches the provided MIME type.
+ *
+ * @since n.e.x.t
+ *
+ * @param int    $attachment_id The ID of the attachment.
+ * @param string $src           The original image src url.
+ * @param string $mime          A mime type we are looking to get image url.
+ * @return string|false Returns mime type image if available, false otherwise.
+ */
+function webp_uploads_get_mime_type_image( int $attachment_id, string $src, string $mime ) {
+	$metadata     = wp_get_attachment_metadata( $attachment_id );
+	$src_basename = wp_basename( $src );
+	if ( isset( $metadata['sources'][ $mime ]['file'] ) ) {
+		$basename = wp_basename( $metadata['file'] );
+
+		if ( $src_basename === $basename ) {
```

Comment:
@westonruter No. When we upload JPEG image the function gets the `foo.jpg` not `foo-jpg.avif`.

You can check it through:
- Add `add_filter( 'wp_calculate_image_sizes', '__return_false' );` in functions.php
- Upload JPEG image, Make sure to enabled the fallback JPEG if not then it will return IMG per #1447
- This function get the JPEG image and it return it's alternate mime type
- Use Full size image then this  condition passes. Like. `foo.jpg === foo.jpg`

<details>

<summary>Image Metadata</summary>

```php
Array
(
    [width] => 1080
    [height] => 608
    [file] => 2024/08/car.jpeg
    [filesize] => 150444
    [sizes] => Array
        (
            [medium] => Array
                (
                    [file] => car-300x169.jpeg
                    [width] => 300
                    [height] => 169
                    [mime-type] => image/jpeg
                    [filesize] => 12915
                    [sources] => Array
                        (
                            [image/jpeg] => Array
                                (
                                    [file] => car-300x169.jpeg
                                    [filesize] => 12915
                                )

                            [image/webp] => Array
                                (
                                    [file] => car-300x169-jpeg.webp
                                    [filesize] => 9580
                                )

                        )

                )

            [large] => Array
                (
                    [file] => car-1024x576.jpeg
                    [width] => 1024
                    [height] => 576
                    [mime-type] => image/jpeg
                    [filesize] => 110947
                    [sources] => Array
                        (
                            [image/jpeg] => Array
                                (
                                    [file] => car-1024x576.jpeg
                                    [filesize] => 110947
                                )

                            [image/webp] => Array
                                (
                                    [file] => car-1024x576-jpeg.webp
                                    [filesize] => 80386
                                )

                        )

                )

            [thumbnail] => Array
                (
                    [file] => car-150x150.jpeg
                    [width] => 150
                    [height] => 150
                    [mime-type] => image/jpeg
                    [filesize] => 6342
                    [sources] => Array
                        (
                            [image/jpeg] => Array
                                (
                                    [file] => car-150x150.jpeg
                                    [filesize] => 6342
                                )

                            [image/webp] => Array
                                (
                                    [file] => car-150x150-jpeg.webp
                                    [filesize] => 4834
                                )

                        )

                )

            [medium_large] => Array
                (
                    [file] => car-768x432.jpeg
                    [width] => 768
                    [height] => 432
                    [mime-type] => image/jpeg
                    [filesize] => 65538
                    [sources] => Array
                        (
                            [image/jpeg] => Array
                                (
                                    [file] => car-768x432.jpeg
                                    [filesize] => 65538
                                )

                            [image/webp] => Array
                                (
                                    [file] => car-768x432-jpeg.webp
                                    [filesize] => 48900
                                )

                        )

                )

        )

    [image_meta] => Array
        (
            [aperture] => 0
            [credit] => 
            [camera] => 
            [caption] => 
            [created_timestamp] => 0
            [copyright] => 
            [focal_length] => 0
            [iso] => 0
            [shutter_speed] => 0
            [title] => 
            [orientation] => 0
            [keywords] => Array
                (
                )

        )

    [dominant_color] => 575757
    [has_transparency] => 
    [sources] => Array
        (
            [image/jpeg] => Array
                (
                    [file] => car.jpeg
                    [filesize] => 150444
                )

            [image/webp] => Array
                (
                    [file] => car-jpeg.webp
                    [filesize] => 109568
                )

        )

)
```
</details>

> Also, the second parameter isn't passed to wp_basename() so the extension isn't getting stripped.

We can't second parameter as we don't know which image gets uploaded. In future we are adding PNG to WebP support. We leverage the code from [webp_uploads_img_tag_update_mime_type()](https://github.com/WordPress/performance/blob/trunk/plugins/webp-uploads/hooks.php#L591) in which we don't use second parameter.